src/opentelemetry.ts

```diff
@@ -33,3 +33,16 @@ export const opentelemetrySDK = new NodeSDK({
   instrumentations: [...getNodeAutoInstrumentations()]
 });
 opentelemetrySDK.start();
+
+process.on('beforeExit', async () => {
+  await Promise.all(
+    Object.entries(opentelemetrySDK)
+      .filter(([_, value]) => value && typeof value.forceFlush === 'function')
+      .map(([name, value]) => {
+        // eslint-disable-next-line no-console
+        console.log(`OpenTelemetry: cleanup ${name}`);
+        return value.forceFlush.call(value);
+      })
+  );
```

Comment:
This should be removed as it is most likely unnecessary, call to `shutdown` appears to handle this exact thing, e.g. here for the traces https://github.com/open-telemetry/opentelemetry-js/blob/90afa2850c0690f7a18ecc511c04927a3183490b/packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts#L108.

Reply:
We need to do it manually; otherwise, I would not raise this PR.

Reply:
The important thing in this PR seems to be the shutdown call. Have you tested the code with just that?

Reply:
We need to call `forceFlush` on all classes, and the problem is that not all of them implement this shutdown functionality as mentioned in `BatchSpanProcesssorBase`.

Reply:
Which ones are problematic? We shouldn't implement a patch around shutdown without pinpointing the bug in the library.

Reply:
It seems it is not needed anymore.